packages/wallet/src/W3mFrameProvider.ts

```diff
@@ -451,15 +462,36 @@ export class W3mFrameProvider {
   }
 
   private async appEvent<T extends W3mFrameTypes.ProviderRequestType>(
-    event: Omit<W3mFrameTypes.AppEvent, 'id'>
+    event: AppEventType
   ): Promise<W3mFrameTypes.Responses[`Frame${T}Response`]> {
     await this.w3mFrame.frameLoadPromise
-    const type = event.type.replace('@w3m-app/', '')
+    let timer: NodeJS.Timeout | undefined = undefined
+
+    function replaceEventType(type: AppEventType['type']) {
+      return type.replace('@w3m-app/', '')
+    }
+
+    const type = replaceEventType(event.type)
+
+    const shouldCheckForTimeout = [
+      W3mFrameConstants.APP_CONNECT_EMAIL,
+      W3mFrameConstants.APP_CONNECT_DEVICE,
+      W3mFrameConstants.APP_CONNECT_OTP,
+      W3mFrameConstants.APP_CONNECT_SOCIAL,
+      W3mFrameConstants.APP_GET_SOCIAL_REDIRECT_URI,
+      W3mFrameConstants.APP_GET_FARCASTER_URI
+    ]
+      .map(replaceEventType)
+      .includes(type)
+
+    if (shouldCheckForTimeout && this.onTimeout) {
+      // 15 seconds timeout
+      timer = setTimeout(this.onTimeout, 15_000)
+    }
```

Comment:
There are some events that don't respond at all, so it wasn't a good idea to have a timeout for all of those. 

Instead i added the necessary ones + i added a 15 second timeout before showing the alert UI.